pkg/provider/k8s/k8s.go

```diff
@@ -36,7 +42,8 @@ type Resource struct {
 
 // K8s holds the fields used to generate API reqeust from within a cluster.
 type K8s struct {
-	clt *kubernetes.Clientset
+	clt       *kubernetes.Clientset
+	apiextclt *apiServerExtensionsClient.Clientset
```

Comment:
Please use camel case to make it easyer to read.
```suggestion
	ApiExtClient *apiServerExtensionsClient.Clientset
```